File: Sources/TokenGauge/UsagePalette.swift
import SwiftUI
import AppKit

/// The single definition of how much-is-left maps to a color, shared by the
/// large panel, the small panel, and the menu bar readout. Previously each of
/// the three carried its own copy, which is how they drifted apart.
enum UsagePalette {
    enum Level {
        case plenty
        case caution
        case low
        case unknown
    }

    static func level(remaining: Double?) -> Level {
        guard let remaining else { return .unknown }
        switch remaining {
        case 50...: return .plenty
        case 15..<50: return .caution
        default: return .low
        }
    }

    static func word(remaining: Double?) -> String {
        switch level(remaining: remaining) {
        case .plenty: return "여유"
        case .caution: return "주의"
        case .low: return "부족"
        case .unknown: return "—"
        }
    }

    // Caution and low are fixed values so they read identically everywhere.
    // Plenty is not: on the panel it's white against dark material, but in the
    // menu bar it has to follow the menu bar, so each caller supplies its own.
    private static let cautionRGB = (red: 0.96, green: 0.86, blue: 0.55) // pastel yellow #F5DB8C
    // Muted to sit in the same family as the yellow and the Claude orange,
    // but kept saturated enough to still read as a warning — going any deeper
    // makes it hard to pick out from the orange bar it sits on top of.
    private static let lowRGB = (red: 0.867, green: 0.310, blue: 0.271) // #DD4F45

    /// - Parameter plenty: the color to use when there's plenty left.
    static func color(remaining: Double?, plenty: Color) -> Color {
        switch level(remaining: remaining) {
        case .plenty: return plenty
        case .caution: return Color(red: cautionRGB.red, green: cautionRGB.green, blue: cautionRGB.blue)
        case .low: return Color(red: lowRGB.red, green: lowRGB.green, blue: lowRGB.blue)
        case .unknown: return .gray
        }
    }

    static func nsColor(remaining: Double?, plenty: NSColor) -> NSColor {
        switch level(remaining: remaining) {
        case .plenty: return plenty
        case .caution: return NSColor(srgbRed: cautionRGB.red, green: cautionRGB.green, blue: cautionRGB.blue, alpha: 1)
        case .low: return NSColor(srgbRed: lowRGB.red, green: lowRGB.green, blue: lowRGB.blue, alpha: 1)
        case .unknown: return plenty
        }
    }
}

File: windows/PotatokenHub/UsagePalette.cs
using System.Windows.Media;

namespace PotatokenHub;

/// <summary>
/// The single definition of how much-is-left maps to a color, shared by the
/// panel and the tray badge — same thresholds and same values as the macOS build.
/// </summary>
public static class UsagePalette
{
    public enum Level
    {
        Plenty,
        Caution,
        Low,
        Unknown,
    }

    public static Level LevelOf(double? remaining) => remaining switch
    {
        null => Level.Unknown,
        >= 50 => Level.Plenty,
        >= 15 => Level.Caution,
        _ => Level.Low,
    };

    // pastel yellow #F5DB8C
    public static readonly Color Caution = Color.FromRgb(0xF5, 0xDB, 0x8C);
    // Muted to sit in the same family as the yellow and the Claude orange, but
    // kept saturated enough to still read as a warning.
    public static readonly Color Low = Color.FromRgb(0xDD, 0x4F, 0x45);

    /// <param name="plenty">The color to use when there's plenty left.</param>
    public static Color ColorFor(double? remaining, Color plenty) => LevelOf(remaining) switch
    {
        Level.Plenty => plenty,
        Level.Caution => Caution,
        Level.Low => Low,
        _ => Colors.Gray,
    };

    // Claude's fixed orange; Codex's neutral gray, picked for a dark tray/panel.
    public static readonly Color ClaudeAccent = Color.FromRgb(0xD9, 0x77, 0x57);
    public static readonly Color CodexAccent = Color.FromRgb(0xBD, 0xBD, 0xC4);
    // A deeper neutral used only for Codex's weekly fill, so the white
    // five-hour fill remains immediately distinguishable where they overlap.
    public static readonly Color CodexWeeklyGauge = Color.FromRgb(0x86, 0x86, 0x90);

    public static Color AccentFor(Core.Provider provider) =>
        provider == Core.Provider.Claude ? ClaudeAccent : CodexAccent;
}
